### PR TITLE
Remove feature gate form `EVM` integration tests.

### LINKED ISSUE #844: Bank integration test in demo-rollup fails if EVM is not feature gated

### PR DESCRIPTION
**Pull request by bkolad**
# Description
Remove feature gate form `EVM` integration tests.

## Linked Issues
- Fixes #844

## Testing
CI passes.

## Docs
No changes.

### CODE REVIEW

File: examples/demo-rollup/tests/all_tests.rs [hunk 1]
```diff
@@ -1,6 +1,3 @@
-// For now, exclude the evm from the bank tests
-#[cfg(not(feature = "experimental"))]
 mod bank;
-#[cfg(feature = "experimental")]
```
Comment (LukaszRozmej) [inline examples/demo-rollup/tests/all_tests.rs:4]: this flags should stay as they were, but CI should get `experimental` feature flag when invoking `cargo llvm-cov --locked --lcov --output-path lcov.info`

Comment (bkolad) [reply examples/demo-rollup/tests/all_tests.rs:4]: We used to import `sov-demo-rollup = { path = ".", features = ["experimental"] }` in tests, but it was removed with #859. I am adding it back as the underlying issue is fixed. cc @preston-evans98

Comment (LukaszRozmej) [reply examples/demo-rollup/tests/all_tests.rs:4]: going to approve, but I still think they should be feature-gated, maybe by `sov-demo-rollup/experimental` feature as they are dependent on that.

File: examples/demo-rollup/Cargo.toml [hunk 2]
```diff
@@ -67,6 +67,7 @@ ethers-signers = { workspace = true }
 ethers = { workspace = true }
 revm = { workspace = true }
 
+sov-demo-rollup = { path = ".", features = ["experimental"] }
```
Comment (LukaszRozmej) [inline examples/demo-rollup/Cargo.toml:70]: was this removed only because of issue in evm?

**Review by LukaszRozmej — APPROVED** [review]

File: examples/demo-rollup/tests/all_tests.rs (hunk 1 above)
Comment (bkolad) [reply examples/demo-rollup/tests/all_tests.rs:4]: I believe it is more practical to have the tests for the `evm` enabled as the default. This  allows other developers to quickly detect any issues they may have caused without having to rely on the CI. Additionally, while it's acceptable to have incomplete documentation for `experimental` features, if someone has taken the time to write a test, we should make sure it functions correctly under all conditions.

**Review by preston-evans98 — APPROVED** [review]

File: examples/demo-rollup/Cargo.toml (hunk 2 above)
Comment (preston-evans98) [reply examples/demo-rollup/Cargo.toml:70]: Yep, that's correct

**Pull request merged**